src/test/coinscachepair_tests.cpp

```diff
@@ -146,70 +146,48 @@ BOOST_AUTO_TEST_CASE(linked_list_add_flags)
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), CCoinsCacheEntry::DIRTY);
+    n1.second.SetDirty(n1, sentinel);
+    BOOST_CHECK(n1.second.IsDirty());
     BOOST_CHECK_EQUAL(n1.second.Next(), &sentinel);
     BOOST_CHECK_EQUAL(n1.second.Prev(), &sentinel);
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n1);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n1);
 
     // Check that adding FRESH flag on new node inserts it after n1
-    n2.second.AddFlags(CCoinsCacheEntry::FRESH, n2, sentinel);
-    BOOST_CHECK_EQUAL(n2.second.GetFlags(), CCoinsCacheEntry::FRESH);
+    n2.second.SetFresh(n2, sentinel);
+    BOOST_CHECK(n2.second.IsFresh());
     BOOST_CHECK_EQUAL(n2.second.Next(), &sentinel);
     BOOST_CHECK_EQUAL(n2.second.Prev(), &n1);
     BOOST_CHECK_EQUAL(n1.second.Next(), &n2);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n2);
 
-    // Check that adding 0 flag has no effect, and doesn't change position
-    n1.second.AddFlags(0, n1, sentinel);
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), CCoinsCacheEntry::DIRTY);
-    BOOST_CHECK_EQUAL(n1.second.Next(), &n2);
-    BOOST_CHECK_EQUAL(n1.second.Prev(), &sentinel);
-    BOOST_CHECK_EQUAL(sentinel.second.Next(), &n1);
-    BOOST_CHECK_EQUAL(n2.second.Prev(), &n1);
-
     // Check that we can add extra flags, but they don't change our position
-    n1.second.AddFlags(CCoinsCacheEntry::FRESH, n1, sentinel);
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), CCoinsCacheEntry::DIRTY | CCoinsCacheEntry::FRESH);
+    n1.second.SetFresh(n1, sentinel);
+    BOOST_CHECK(n1.second.IsDirty() && n1.second.IsFresh());
     BOOST_CHECK_EQUAL(n1.second.Next(), &n2);
     BOOST_CHECK_EQUAL(n1.second.Prev(), &sentinel);
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n1);
     BOOST_CHECK_EQUAL(n2.second.Prev(), &n1);
 
     // Check that we can clear flags then re-add them
     n1.second.ClearFlags();
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), 0);
+    BOOST_CHECK(!n1.second.IsDirty() && !n1.second.IsFresh());
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n2);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n2);
     BOOST_CHECK_EQUAL(n2.second.Next(), &sentinel);
     BOOST_CHECK_EQUAL(n2.second.Prev(), &sentinel);
 
-    // Check that calling `ClearFlags` with 0 flags has no effect
     n1.second.ClearFlags();
-    BOOST_CHECK_EQUAL(n1.second.GetFlags(), 0);
-    BOOST_CHECK_EQUAL(sentinel.second.Next(), &n2);
-    BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n2);
-    BOOST_CHECK_EQUAL(n2.second.Next(), &sentinel);
-    BOOST_CHECK_EQUAL(n2.second.Prev(), &sentinel);
-
-    // Adding 0 still has no effect
-    n1.second.AddFlags(0, n1, sentinel);
+    BOOST_CHECK(!n1.second.IsDirty() && !n1.second.IsFresh());
     BOOST_CHECK_EQUAL(sentinel.second.Next(), &n2);
     BOOST_CHECK_EQUAL(sentinel.second.Prev(), &n2);
     BOOST_CHECK_EQUAL(n2.second.Next(), &sentinel);
     BOOST_CHECK_EQUAL(n2.second.Prev(), &sentinel);
 
     // But adding DIRTY re-inserts it after n2
```

Comment:
`But` can be removed now.

Reply:
Because class declarations don't need semicolons, but I'll revert if you find it distracting

Reply:
Done

Reply:
Was already removed in `coins: Remove direct GetFlags access`, but I've moved it up to `coins, refactor: Split up AddFlags to remove invalid states`